api/src/main/java/org/openmrs/module/fhir2/api/util/FhirUtils.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.util;
+
+import javax.validation.constraints.NotNull;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apache.commons.lang.StringUtils;
+import org.hl7.fhir.r4.model.Identifier;
+import org.hl7.fhir.r4.model.Reference;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class FhirUtils {
+	
+	public static Reference addReference(@NotNull String uri, String givenName, String familyName, String identifier) {
+		Reference reference = new Reference();
+		Identifier fhirIdentifier = new Identifier();
+		fhirIdentifier.setValue(identifier);
+		
+		StringBuilder displayName = new StringBuilder();
```

Comment:
Instead of doing the formatting here, we should just pass the result of `PersonName#getFullName()` into this method and use that as the display name.